### PR TITLE
[terra-infinite-list] Update documentation for headers and subheaders

### LINKED ISSUE #512: Update Infinite List doc for SectionHeader and SubsectionHeader usage
# Feature Request
Update Infinite List doc for SectionHeader and SubsectionHeader usage

## Description
Add documentation similar to the list guides to use the SectionHeader and SubsectionHeader within the infinite list, specfically to steer consumers away from accidentally using a section.

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Add guides for assembling infinite list content with section headers and subsection headers.

Resolves #512 

Thanks for contributing to Terra.
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-infinite-list/docs/InfiniteListSection.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const section = [
+      <SectionHeader
+        key={sectionData.key}
+        title={sectionData.title}
+      />,
+    ];
+    return section;
+  };
+```
+Next we'll loop through the associated child items for the section appending them to the initial array and then returning the child array prepended the generated array of items to the section header.
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/docs/InfiniteListSection.md]: I'm a little confused by this statement

File: packages/terra-infinite-list/docs/InfiniteListSection.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Terra InfiniteList - Sections
+
+To acheive sections within the InfiniteList the only supported implementation is directly consuming the static variant of the `SectionHeader` component. Expand/collapse functionality is not supported within the infinite list.
```
Comment (bjankord) [inline packages/terra-infinite-list/docs/InfiniteListSection.md]: typo `acheive` should be `achieve`

File: packages/terra-infinite-list/docs/InfiniteListSubsection.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# Terra InfiniteList - Subsections
+
+To acheive sections within the InfiniteList the only supported implementation is directly consuming the static variant of the `SectionHeader` and `SubsectionHeader` component. Expand/collapse functionality is not supported within the infinite list.
```
Comment (bjankord) [inline packages/terra-infinite-list/docs/InfiniteListSubsection.md]: typo `acheive` should be `achieve`

File: packages/terra-infinite-list/docs/InfiniteListSection.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Terra InfiniteList - Sections
+
+To acheive sections within the InfiniteList the only supported implementation is directly consuming the static variant of the `SectionHeader` component. Expand/collapse functionality is not supported within the infinite list. 
+
+## Map Section Structure
+So in our first step a `SectionHeader` needs to be created for each associated section. We'll add it to the initialization of an array that will be expanded in following steps.
```
Comment (yuderekyu) [inline packages/terra-infinite-list/docs/InfiniteListSection.md]: expanded in *the following

File: packages/terra-infinite-list/docs/ProgressiveLoadingList.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# Terra InfiniteList - Progressive Loading
+
+To limit the impact of large data sets, progressive loading of list items is available through the InfiniteList. As the InfiniteList approaches the bottom of the it's scrollHeight the `onRequestItems` callback will execute, allowing the implementation to lazy load data into the list.
```
Comment (yuderekyu) [inline packages/terra-infinite-list/docs/ProgressiveLoadingList.md]: of *it's

File: packages/terra-infinite-list/docs/ProgressiveLoadingList.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+```jsx
+  createItems() {
+    return this.state.items.map(item => (
+      <Item key={item.key}>
+        <Placeholder title={item.title} />
+      </Item>
+    ));
+  }
+```
+Finally we'll declare our infinite list looping. Providing the InfiniteList with an `initialLoadingIndicator` and `progressiveLoadingIndicator`. The infinite list doesn't have no visual styling, so loading indicators are customizable. Followed up with setting the `onRequestItems` to our requestData method.
```
Comment (yuderekyu) [inline packages/terra-infinite-list/docs/ProgressiveLoadingList.md]: *does not have visual styling?

File: packages/terra-infinite-list/docs/InfiniteListSection.md (hunk 1 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/docs/InfiniteListSection.md]: Updated.

File: packages/terra-infinite-list/docs/InfiniteListSection.md (hunk 2 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/docs/InfiniteListSection.md]: Updated.

File: packages/terra-infinite-list/docs/InfiniteListSubsection.md (hunk 3 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/docs/InfiniteListSubsection.md]: Updated.

File: packages/terra-infinite-list/docs/InfiniteListSection.md (hunk 4 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/docs/InfiniteListSection.md]: Updated.

File: packages/terra-infinite-list/docs/ProgressiveLoadingList.md (hunk 5 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/docs/ProgressiveLoadingList.md]: Updated.

File: packages/terra-infinite-list/docs/ProgressiveLoadingList.md (hunk 6 above)
Comment (dkasper-was-taken) [reply packages/terra-infinite-list/docs/ProgressiveLoadingList.md]: Updated.

**Pull request merged**